### PR TITLE
Add support in tipseen component for: submitButtonText & submitButtonOnClick & dismissButtonText & dismissButtonOnClick & hideWhenReferenceHidden

### PR DESCRIPTION
**Pull request by hadasfa**
## Go over this checklist before submitting your Pull Request

Description: Add support in tipseen component for: submitButtonText & submitButtonOnClick & dismissButtonText & dismissButtonOnClick & hideWhenReferenceHidden.
The change in the button props is moshe tzemach idea after reviewing the tipseen in monolith changes.
The hideWhenReferenceHidden support is for fixing the monolith bug.

#### Basic
- [v] PR has description
- [ ] Changes to the component are backward compatible (including selectors structure). If not - add to the title of the PR "BREAKABLE_CHANGE""
- [v] All changes to the component are reflected in the ReadMe
- [ ] If component is old and was not compliant with the latest guidelines - it was fixed (optional) 
#### Style
- [ ] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [v] All the changes to the component should be reflected in the Storybook.
- [v] Component passed Accessibility Plugin checks
#### Tests
- [v] All current tests are passing
- [v] New functionality is covered with tests
- [v] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

File: src/components/Tipseen/__stories__/tipseen.stories.js [hunk 1]
```diff
@@ -94,11 +94,14 @@ export const Sandbox = () => (
...
       justify={select("justify", Tipseen.justifyTypes, Tipseen.justifyTypes.CENTER)}
       isCloseButtonHidden={boolean("Is close button hidden", false)}
+      hideWhenReferenceHidden={boolean("Hide when reference hidden", false)}
       content={
         <TipseenContent
           title={text("Title", "Title")}
-          isDismissHidden={boolean("Is dismiss hidden", false)}
+          isDismissHidden={boolean("Is dismiss hidden", true)}
           children={text("Tipseen content text", "Popover message will appear here loremipsum dolor samet…")}
+          submitButtonText={text("Submit button text", "Submit button text")}
```
Comment (hadasfa) [inline src/components/Tipseen/__stories__/tipseen.stories.js:103]: In evginy design the button called "submit" but in the monolith it's standard name is got it.
so, here the api remain the same but in the monolith I am using tipseen content wrapper which expect the receive "gotItButtonText" and "gotItCallback" (which there I also close the tipseen with the redux monolith mechanism).
I think that "gotItButtonText" and "gotItCallback" will be little bit weird inside an open source library api. what do you think?

File: src/components/Tipseen/TipseenContent.jsx [hunk 2]
```diff
@@ -15,11 +16,38 @@ const TipseenContent = ({
...
+  dismissButtonText,
+  onDismiss,
+  // Backward compatibility for props naming
   dismissButtonProps,
+  // Backward compatibility for props naming
   submitButtonProps
 }) => {
-  const { content: dismissContent, className: dismissClassName, ...otherDismissButtonProps } = dismissButtonProps;
-  const { content: submitContent, className: submitClassName, ...otherSubmitButtonProps } = submitButtonProps;
+  debugger;
```
Comment (orrgottlieb) [inline src/components/Tipseen/TipseenContent.jsx:28]: remove debugger;

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/helpers/backwardCompatibilityForProperties.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
-export function backwardCompatibilityForProperties(valuesArrayByMostUpdateNaming) {
-  return valuesArrayByMostUpdateNaming.find(value => value !== undefined);
+export function backwardCompatibilityForProperties(valuesArrayByMostUpdateNaming, defaultValue) {
```
Comment (orrgottlieb) [inline src/helpers/backwardCompatibilityForProperties.js:1]: 1. please add defaults to the arguments `EMPTY_ARRAY`
2. Please consider adding useMemo on the spread of the array which is being passed.

File: src/components/Tipseen/TipseenContent.jsx [hunk 4]
```diff
@@ -15,11 +16,37 @@ const TipseenContent = ({
...
+    [dismissButtonText, dismissContent],
+    DISMISS_BUTTON_TEXT
+  );
+  const overrideDismissOnClick = backwardCompatibilityForProperties([onDismiss, dismissDeprecatedOnClick], NOOP);
+  const {
+    content: submitContent,
+    className: submitClassName,
+    onClick: submitDeprecatedOnClick,
+    ...otherSubmitButtonProps
+  } = submitButtonProps || {};
```
Comment (orrgottlieb) [inline src/components/Tipseen/TipseenContent.jsx:44]: EMOTY_OBJECT

File: src/components/Tipseen/TipseenContent.jsx [hunk 5]
```diff
@@ -15,11 +16,37 @@ const TipseenContent = ({
...
   submitButtonProps
 }) => {
-  const { content: dismissContent, className: dismissClassName, ...otherDismissButtonProps } = dismissButtonProps;
-  const { content: submitContent, className: submitClassName, ...otherSubmitButtonProps } = submitButtonProps;
+  const {
+    content: dismissContent,
+    className: dismissClassName,
+    onClick: dismissDeprecatedOnClick,
+    ...otherDismissButtonProps
+  } = dismissButtonProps || {};
```
Comment (orrgottlieb) [inline src/components/Tipseen/TipseenContent.jsx:33]: EMOTY_OBJECT

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**